transaction.go

```diff
@@ -162,6 +162,30 @@ func (t *Transaction) getByRelationship(relationship, relationshipID []byte, ent
 	return
 }
 
+func (t *Transaction) getFiltered(seekTo []byte, entries reflect.Value, limit int64, filters []Filter) (err error) {
+	if limit == 0 {
+		return
+	}
+
+	var count int64
+	err = t.forEachID(seekTo, func(entryID []byte) (err error) {
+		val := t.c.newEntryValue()
+		if err = t.get(entryID, &val); err != nil {
+			return
+		}
+
+		entries.Set(reflect.Append(entries, reflect.ValueOf(val)))
```

Comment:
I suspect this will take some performance out of this puppy. Just a lil sus haha

Reply:
It's concerning that we are reflecting all the values.. Honestly need a benchmark, because all I know is that it's "slow" from others, but a benchmark would shed light on how slow.
However, if we were more rigid with DBs and had a defined type in gdbu this could be free perf gain.

Reply:
This is how we do GetByRelationship currently. Until we get generics, there is no way to handle this :(. This is how json unmarshalling works under the hood as well.

I'll add a benchmark for this vs ForEach with manual appending